src/components/select-custom/select-custom.component.ts

```diff
@@ -0,0 +1,128 @@
+import { Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
+
+@Component({
+    selector: 'ux-select-custom',
+    templateUrl: './select-custom.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            multi: true,
+            useExisting: forwardRef(() => SelectCustomComponent)
+        }
+    ],
+    host: {
+        '[style.width.px]': 'width'
+    }
+})
+export class SelectCustomComponent<T> implements ControlValueAccessor, OnChanges, OnInit, OnDestroy {
+
+    /** Define the selected item */
+    @Input() selected: T;
+
+    /** Hide the filter input */
+    @Input() hideFilter: boolean;
+
+    /** Debounce the filter change event emitter */
+    @Input() filterChangeDebounce: number = 500;
+
+    /** Define the width of the component */
+    @Input() width: number;
+
+    /** Define the max height of the dropdown */
+    @Input() maxHeight: number;
+
+    /** Determine if we should show the loading indicator */
+    @Input() busy: boolean;
+
+    /** Define if null values are allowed */
+    @Input() allowNull: boolean;
+
+    /** Define the placeholder for the filter input */
+    @Input() placeholder: string = 'Type to filter...';
+
+    /** Emit when the selected item is changed */
+    @Output() selectedChange = new EventEmitter<T>();
+
+    /** Emit when the filter text is changed */
+    @Output() filterChange = new EventEmitter<string>();
+
+    @ContentChild('dropdownContent', { static: false }) dropdownContentRef: TemplateRef<void>;
+    @ContentChild('buttonContent', { static: false }) buttonContentRef: TemplateRef<void>;
+
+    @ViewChild('filterInput', { static: false }) filterInputElement: ElementRef;
+
+    filterText: string = '';
+    filterTextChanged$: Subject<string> = new Subject<string>();
+
+    onChange: (_: T) => void = () => { };
+    onTouched: () => void = () => { };
+
+    private readonly _onDestroy$ = new Subject<void>();
+
+    constructor(private readonly _elementRef: ElementRef) { }
+
+    ngOnInit(): void {
+        this.filterTextChanged$.pipe(
+            debounceTime(this.filterChangeDebounce),
+            distinctUntilChanged(), // only emit if value is different from previous value
+            takeUntil(this._onDestroy$)
+        ).subscribe(textValue => {
+            this.filterText = textValue;
+            this.filterChange.emit(this.filterText);
+        });
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.selected) {
+            this.onChange(changes.selected.currentValue);
+            this.onTouched();
+        }
+    }
+
+    ngOnDestroy(): void {
+        this._onDestroy$.next();
+        this._onDestroy$.complete();
+    }
+
+    onToggle(): void {
+        setTimeout(
+            () => {
+                if (this._elementRef.nativeElement.classList.contains('open')) {
```

Comment:
Is this going to have any effect? From what I recall the open class was added by the ngx-bootstrap dropdown but not by ux-menu, so perhaps this isn't needed? And if it is needed perhaps if it is needed we could just use the `opened` event triggered by the `ux-menu` component to avoid having to check whether it is opened or closed.

Reply:
I overlooked it when I removed the bootstrap code. I had already fixed this in my previous commit (exactly in the way you described).